src/main/java/uk/gov/justice/probation/courtcaseservice/service/DocumentTypeFilter.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.justice.probation.courtcaseservice.service;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.justice.probation.courtcaseservice.service.model.KeyValue;
+import uk.gov.justice.probation.courtcaseservice.service.model.document.DocumentType;
+import uk.gov.justice.probation.courtcaseservice.service.model.document.OffenderDocumentDetail;
+
+@Component
+public class DocumentTypeFilter implements Predicate<OffenderDocumentDetail> {
+
+    private final Set<DocumentType> documentTypes;
+
+    private final Set<String> subTypeCodes;
+
+    public DocumentTypeFilter(@Value("#{'${community-api.filter.document.types}'.split(',')}") List<DocumentType> documentTypes,
```

Comment:
Might be worth being more specific about the purpose of the filter in the property. We're going to be making other calls to the documents endpoint to get NSI docs for the `breach` endpoint at a later time so that could cause some confusion if we're not specific about what this filter is used for.

Something like `community-api.filters.probation-record-filter.document.types` maybe?

Wondering whether it would be worth standardising our nomenclature a bit more as well. We refer to `offender` in a number of places and then it's returned as the `probation-record` resource, it's got quite a bit more than just details of the offender by now, think that name is a bit of a hangover.

Reply:
I have renamed the properties. Yes - it's confusing but everything is related to the offender but maybe not the "probation" !

Reply:
Renamed Offender to ProbationRecord and associated methods